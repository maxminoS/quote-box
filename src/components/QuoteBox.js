import React, { useState, useEffect } from "react";
import { Button } from "@material-ui/core";
import { Card, CardContent, CardActions } from "@material-ui/core";

import Quote from "components/Quote";
import Share from "components/Share";
import QuoteApi from "api/QuoteApi";

export default () => {
  const [quotes, setQuotes] = useState([]);
  const [quoteNumber, setQuoteNumber] = useState(0);

  useEffect(async () => {
    QuoteApi.getQuote()
      .then(quotes => {
        setQuotes(quotes);
        setQuoteNumber(Math.floor(Math.random() * quotes.length));
      });
  }, []);

  const handleClick = () => {
    setQuoteNumber(Math.floor(Math.random() * quotes.length));
  };

  return (
    <Card raised="true">
      <CardContent>
        <Quote
          quote={quotes[quoteNumber]}
        />
      </CardContent>
      <CardActions>
        <Share
          quote={quotes[quoteNumber]}
        />
        <Button variant="contained" color="secondary" style={{marginLeft: 'auto'}} onClick={handleClick}>
          GET
        </Button>
      </CardActions>
    </Card>
  );
}
