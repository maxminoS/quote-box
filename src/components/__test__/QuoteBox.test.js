import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import QuoteBox from 'components/QuoteBox';
import QuoteApi from "api/QuoteApi";

jest.mock('api/QuoteApi');

it('renders Quote component correctly and GET button works', async () => {
  jest.spyOn(QuoteApi, 'getQuote');
  QuoteApi.getQuote.mockImplementation(() => Promise.resolve([{ "quote": "a quote", "author": "by this"}]));

  render(<QuoteBox />);
  expect(QuoteApi.getQuote).toHaveReturnedTimes(1);
  expect(await screen.findByRole('heading', { name: /a quote/i })).toBeInTheDocument();
  expect(await screen.findByRole('heading', { name: /\- by this/i })).toBeInTheDocument();

  userEvent.click(screen.getByRole('button', { name: /get/i }));
  QuoteApi.getQuote.mockRestore();
});

it('renders QuoteBox component as expected', () => {
  const quoteBox = renderer.create(<QuoteBox />).toJSON();
  expect(quoteBox).toMatchSnapshot();
});
