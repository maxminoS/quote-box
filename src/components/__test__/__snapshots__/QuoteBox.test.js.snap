// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QuoteBox component as expected 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation8 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": "auto",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        GET
      </span>
    </button>
  </div>
</div>
`;
